### PR TITLE
Fix some complexity in the call() method

### PR DESCRIPTION
**Pull request by borgogelli**
I have tried to remove some of the complexity of the call() method in the Resource class

### CODE REVIEW

**Comment by borgogelli** [comment]
I signed it

**Comment by borgogelli** [comment]
I signed it

File: src/Google/Service/Resource.php [hunk 1]
```diff
@@ -232,6 +200,37 @@ public function call($name, $arguments, $expectedClass = null)
...
 
+  private function logServiceError($txt, $name, $parameter = null)
+  {
+    $array = array(
+        'service' => $this->serviceName,
+        'resource' => $this->resourceName,
+        'method' => $name,
+        'parameter' => $parameter
+    );
+    if ($parameter) {
```
Comment (bshaffer) [inline src/Google/Service/Resource.php]: this conditional is not necessary, please remove it!

File: src/Google/Service/Resource.php [hunk 2]
```diff
@@ -232,6 +200,37 @@ public function call($name, $arguments, $expectedClass = null)
...
+  }
+  
+  private function logServiceInfo($txt, $name, $parameters)
+  {
+    $array  = array(
+            'service' => $this->serviceName,
+            'resource' => $this->resourceName,
+            'method' => $name,
+            'arguments' => $parameters,
+        );
```
Comment (bshaffer) [inline src/Google/Service/Resource.php]: This indentation is off - this should be back four spaces, and the array arguments should only be indented 2 spaces.

File: src/Google/Service/Resource.php [hunk 3]
```diff
@@ -232,6 +200,37 @@ public function call($name, $arguments, $expectedClass = null)
...
+    }
+    $this->client->getLogger()->error(
+        $txt,
+        $array
+    );
+  }
+  
+  private function logServiceInfo($txt, $name, $parameters)
+  {
+    $array  = array(
```
Comment (bshaffer) [inline src/Google/Service/Resource.php]: Let's use a variable name like `$context`, to match with the [PSR spec](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md#3-psrlogloggerinterface).

File: src/Google/Service/Resource.php [hunk 4]
```diff
@@ -232,6 +200,37 @@ public function call($name, $arguments, $expectedClass = null)
     return $this->client->execute($request, $expectedClass);
   }
 
+  private function logServiceError($txt, $name, $parameter = null)
```
Comment (bshaffer) [inline src/Google/Service/Resource.php]: Let's make the method signature look like this:

```
private function log($level, $txt, $name, $parameter = null)
```

Then we can make "info" and "error" the first argument to these calls, and consolidate two methods into one.

**Comment by bshaffer** [comment]
As I have not heard back, closing this PR.

**Pull request closed without merging**